### PR TITLE
Cw1 Whitelist contracts done properly

### PR DESCRIPTION
**Pull request by hashedone**

### CODE REVIEW

File: contracts/cw1-whitelist/src/contract.rs [hunk 1]
```diff
@@ -1,243 +1,451 @@
...
+            mutable: Item::new("mutable"),
         }
     }
 
     #[msg(instantiate)]
     pub fn instantiate(
         &self,
-        (deps, _env, _msg): (DepsMut, Env, MessageInfo),
-        members: Vec<String>,
+        ctx: (DepsMut, Env, MessageInfo),
```
Comment (jawoznia) [inline contracts/cw1-whitelist/src/contract.rs:30]: Maybe just `(deps, _, _): (DepsMut, Env, MessageInfo),`

I know it's not the preatiest approach, but you do it later in line 34 anyway.
Until we will introduce #30 I think we can keep it as it was with only the difference to change `_env` to just `_`

File: contracts/cw1-whitelist/src/contract.rs [hunk 2]
```diff
@@ -1,243 +1,451 @@
...
+        mutable: bool,
     ) -> Result<Response, ContractError> {
-        for addr in members.into_iter() {
-            self.members
-                .save(deps.storage, deps.api.addr_validate(&addr)?, &Empty {})?;
+        let (deps, _, _) = ctx;
+        set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+        for admin in admins {
+            let admin = deps.api.addr_validate(&admin)?;
```
Comment (jawoznia) [inline contracts/cw1-whitelist/src/contract.rs:38]: Just a nitpick to consider. Bot versions fine by me
```
            self.admins
                .save(deps.storage, &deps.api.addr_validate(&admin)?, &Empty {})?;

```

Comment (hashedone) [reply contracts/cw1-whitelist/src/contract.rs:38]: Isn't it the same just inlined?

Comment (jawoznia) [reply contracts/cw1-whitelist/src/contract.rs:38]: It is. This is how it was done earlier so it's just a nitpick. Both versions fine by me

**Pull request merged**